autoscale/src/main/java/com/sequenceiq/periscope/controller/AutoScaleClusterV2Controller.java

```diff
@@ -24,39 +26,56 @@
     @Inject
     private ClusterService clusterService;
 
+    @Inject
+    private AuthorizationService authorizationService;
+
     @Override
     public AutoscaleClusterResponse modifyByCloudbreakCluster(AutoscaleClusterRequest ambariServer, Long stackId) {
-        return autoScaleClusterCommonService.modifyCluster(ambariServer, clusterService.findOneByStackId(stackId).getId());
+        Cluster cluster = clusterService.findOneByStackId(stackId);
+        authorizationService.hasReadPermission(cluster);
+        return autoScaleClusterCommonService.modifyCluster(ambariServer, cluster.getId());
     }
 
     @Override
     public AutoscaleClusterResponse getByCloudbreakCluster(Long stackId) {
-        return autoScaleClusterCommonService.getCluster(clusterService.findOneByStackId(stackId).getId());
+        Cluster cluster = clusterService.findOneByStackId(stackId);
+        authorizationService.hasReadPermission(cluster);
+        return autoScaleClusterCommonService.getCluster(cluster.getId());
     }
 
     @Override
     public void deleteByCloudbreakCluster(Long stackId) {
-        autoScaleClusterCommonService.deleteCluster(clusterService.findOneByStackId(stackId).getId());
+        Cluster cluster = clusterService.findOneByStackId(stackId);
+        authorizationService.hasReadPermission(cluster);
```

Comment:
Is it okay that it's read permission for cluster delete?

Reply:
Thanks good catch, i replaced the original solution without brain :D